parser/parser_test.go

```diff
@@ -484,6 +486,56 @@ func TestLenientParse_OptionNameTrailingDot(t *testing.T) {
 									int32 bar = 1 [baz.(foo)=1];
 								}`,
 		},
+		"field-options-type": {
```

Comment:
How about other (non-compact) options and inside message literals? Like the following cases:

```proto
message Foo {
    option (type.) = "abc";
    option (msgLiteral) = {
        [type.]: <
            [type.googleapis.com/type.]: < a: "b" >
        >
    }
}
```